### ISSUE TITLE
A simple question: AudioPlayerState.COMPLETED, STOPPED, and null?

### ISSUE THREAD

**Issue by fzyzcjy** [luanpotter/audioplayers#647]
Hi thanks for this wonderful lib! When using `AudioPlayerState` I am confused about AudioPlayerState.COMPLETED vs STOPPED. The comments beside it says it is "self-explanatory", but I am not sure about it. So I guess the following from some observations, is it correct?

* `null`: Not started playing? (Not sure, I would appreciate it if could be explained)
* `STOPPED`: Hit `stop()`? (Not sure as well)
* `COMPLETED`: When I play the whole audio and it reaches its end, instead of when people call `stop`? (Not sure as well)

I do know the following: 

* `PLAYING`: of course, audio is playing
* `PAUSED` : of course, when hit `pause()`. The resource is ***NOT*** released, and the position is kept.

Thanks!

**Full Description**
N/A

**Code to Reproduce**
N/A

**Log Errors**
N/A

**Files/URLs/Sources**
N/A

**Screenshots**
N/A

**Platforms**
N/A

**Comment by luanpotter** [comment]
That is a good question. Looking at the code I believe all your observations are correct. The only thing to add is that SOPPED is also set if there is an error preventing it to reach the end.

I updated the doc with some more comments: https://github.com/luanpotter/audioplayers/pull/650

But let me tag @renancaraujo as I believe he built this. I agree the state diagram would be cool.

Some edge cases to think about:

 * should it start with null? we could have IDLE, LOADED states
 * if it's on LOOP mode, does it ever COMPLETED? or stays PLAYING forever?
 * should we have a RELEASED state?
 * should we have a different state for errors or is STOPPED ok?
 * comparing L118 and L637 (on my PR), is this inconsistency a bug?

**Comment by fzyzcjy** [comment]
@luanpotter Thanks! By the way I also found a misleading edge case: When I am playing and directly call `dispose`, it does not throw exception, and the audio is *still playing* (in Android). Naively thinking, it should stop it automatically.

**Comment by luanpotter** [comment]
Yeah that sounds like a bug, but unrelated. Feel free to open another issue for that one with more details so we can investigate (there is a template on gh)
